validator/src/main/java/com/google/daq/mqtt/validator/Validator.java

```diff
@@ -646,10 +653,40 @@ private void handleUdmiConfig(UdmiConfig udmiConfig) {
     JsonUtil.writeFile(udmiConfig, new File(outBaseDir, UDMI_CONFIG_JSON_FILE));
   }
 
+  private boolean handleMetadataUpdate(Map<String, String> attributes, Object messageObject) {
+    String deviceId = attributes.get("deviceId");
+
+    if (!reportingDevices.containsKey(deviceId)) {
+      return false;
+    }
+
+    ReportingDevice device = reportingDevices.get(deviceId);
+    String subFolderRaw = attributes.get(SUBFOLDER_PROPERTY_KEY);
+    String subTypeRaw = attributes.get(SUBTYPE_PROPERTY_KEY);
+
+    if (SubType.MODEL.value().equals(subTypeRaw) && SubFolder.UPDATE.value().equals(subFolderRaw)) {
+      Metadata metadata = convertTo(Metadata.class, messageObject);
+
+      if (metadata == null) {
```

Comment:
This can only happen if `messageObject` is `null` to begin with, so I think it would be better to have a `requireNonNull()` on `messageObject`, since it should never even get here in that case!  (But if it does somehow when it would be very obvious!)